src/App/Fossa/ManualDeps.hs

```diff
@@ -83,9 +86,17 @@ toSourceUnit :: (Has (Lift IO) sig m, Has Diagnostics sig m, Has StickyLogger si
 toSourceUnit root depsFile manualDeps@ManualDependencies{..} maybeApiOpts = do
   -- If the file exists and we have no dependencies to report, that's a failure.
   when (hasNoDeps manualDeps) $ fatalText "No dependencies found in fossa-deps file"
+  -- TODO: archiveOrCLI will eventually be set by querying core to see if we should archive upload or cli-side license scan
+  -- See https://fossa.atlassian.net/browse/ANE-23
+  -- IMPORTANT: Until ANE-23 is implemented, this should always be `ArchiveUpload` on master
+  let archiveOrCLI = ArchiveUpload
   archiveLocators <- case maybeApiOpts of
-    Nothing -> pure $ archiveNoUploadSourceUnit vendoredDependencies
-    Just apiOpts -> archiveUploadSourceUnit root apiOpts vendoredDependencies
+    Nothing -> case archiveOrCLI of
```

Comment:
[Optional] You can pattern match on multiple cases, making it easier to parse 4 branches

```haskell
  -- TODO: archiveOrCLI will eventually be set by querying core to see if we should archive upload or cli-side license scan
  -- See https://fossa.atlassian.net/browse/ANE-23
  -- IMPORTANT: Until ANE-23 is implemented, this should always be `ArchiveUpload` on master
  let defaultArchiveOrCLI = ArchiveUpload

  archiveLocators :: [Locator] <- case (maybeApiOpts, defaultArchiveOrCLI) of
    (Just apiOpts, ArchiveUpload) -> archiveUploadSourceUnit root apiOpts vendoredDependencies
    (Just apiOpts, CLILicenseScan) -> licenseScanSourceUnit root apiOpts vendoredDependencies
    (Nothing, ArchiveUpload) -> pure $ archiveNoUploadSourceUnit vendoredDependencies
    (Nothing, CLILicenseScan) -> pure $ licenseNoScanSourceUnit vendoredDependencies

```